packages/wouter/test/link.test.tsx

```diff
@@ -1,206 +1,221 @@
-});
+    fireEvent.click(getByTestId("link"));
+    expect(location.pathname).not.toBe("/users");
+  });
 
-it("ignores the navigation when clicked with modifiers", () => {
-  const { getByTestId } = render(
-    <Link href="/users" data-testid="link">
-      click
-    </Link>
-  );
-  const clickEvt = new MouseEvent("click", {
-    bubbles: true,
-    cancelable: true,
-    button: 0,
-    ctrlKey: true,
-  });
-
-  // js-dom doesn't implement browser navigation (e.g. changing location
-  // when a link is clicked) so we need just ingore it to avoid warnings
-  clickEvt.preventDefault();
-
-  fireEvent(getByTestId("link"), clickEvt);
-  expect(location.pathname).not.toBe("/users");
-});
+  it("accepts an `onClick` prop, fired before the navigation", () => {
+    const clickHandler = vi.fn();
 
-it("ignores the navigation when event is cancelled", () => {
-  const clickHandler: MouseEventHandler = (e) => {
-    e.preventDefault();
-  };
+    const { getByTestId } = render(
+      <Link href="/" onClick={clickHandler} data-testid="link" />
+    );
 
-  const { getByTestId } = render(
-    <Link href="/users" data-testid="link" onClick={clickHandler}>
-      click
-    </Link>
-  );
+    fireEvent.click(getByTestId("link"));
+    expect(clickHandler).toHaveBeenCalledTimes(1);
+  });
 
-  fireEvent.click(getByTestId("link"));
-  expect(location.pathname).not.toBe("/users");
-});
+  it("renders `href` with basepath", () => {
+    const { getByTestId } = render(
+      <Router base="/app">
+        <Link href="/dashboard" data-testid="link" />
+      </Router>
+    );
 
-it("accepts an `onClick` prop, fired before the navigation", () => {
-  const clickHandler = vi.fn();
+    const link = getByTestId("link");
+    expect(link.getAttribute("href")).toBe("/app/dashboard");
+  });
 
-  const { getByTestId } = render(
-    <Link href="/" onClick={clickHandler}>
-      <a data-testid="link" />
-    </Link>
-  );
+  it("renders `href` with absolute links", () => {
+    const { getByTestId } = render(
+      <Router base="/app">
+        <Link href="~/home" data-testid="link" />
+      </Router>
+    );
 
-  fireEvent.click(getByTestId("link"));
-  expect(clickHandler).toHaveBeenCalledTimes(1);
+    const element = getByTestId("link");
+    expect(element).toHaveAttribute("href", "/home");
+  });
+
+  it("supports history state", () => {
+    const testState = { hello: "world" };
+    const { getByTestId } = render(
+      <Link href="/goo-baz" state={testState} data-testid="link">
+        link
+      </Link>
+    );
+
+    fireEvent.click(getByTestId("link"));
+    expect(location.pathname).toBe("/goo-baz");
+    expect(history.state).toBe(testState);
+  });
 });
 
-it("renders `href` with basepath", () => {
-  const { getByTestId } = render(
-    <Router base="/app">
-      <Link href="/dashboard" data-testid="link" />
-    </Router>
-  );
+describe("<Link /> in asChild mode", () => {
+  it("works for any other elements as well", () => {
```

Comment:
I think the test case label is a bit confusing. How about:

```suggestion
  it("when `asChild` is not specified, wraps the children in an <a />", () => {
```